index.php

```diff
@@ -119,6 +119,7 @@ function failed($reason) {
 
 if (file_exists($testfile)) {
     $status .= "sitedata OK<br>\n";
+    unlink($testfile);
```

Comment:
I guess this depends on what you really want to test.

The original design intent was that index.php is for testing the health of a single front end, while croncheck.php was the health of the system overall. The file names are not the best but it is being used in the wild like this so I don't want to change the semantics of the two files. The difference being that if index.php returns 503 then that frontend may be given less traffic or pulled from the load balancer, vs croncheck is what we wire up to nagios / icinga. Also note that if you have say 30 fronts ends for the same moodle then each of them has their health checked and they may overlap so we don't want race conditions here. This gets worse if you are fast churning a single file on gluster.

We've progressively done less checks on the front ends this way and moved more to the aws tools, because it's too easy to yank a health front end for the wrong reason.

So what we should be testing is the health of the mount to datadir and not the health of the disc itself. If it's out of disc then that's not a reason to pull the front end as you will just scale down to zero and keep spawning new frontends that fail their health check. So this check should instead be in croncheck.php

But this also feels not right, you should have nagios alerts when disc is at say < 5% capacity or whatever, not when the last byte is already gone, it's alerting way too late. You should be alerted like a week before it goes critical, not 1-5 mins afterwards.

Reply:
Croncheck makes a lot more sense here. I'm going to think about exactly what "the site being up" means.